src/GslCore/AstTypes.fs

```diff
@@ -88,30 +88,43 @@ let posBracketTokens left right : SourcePosition = {s = left.pos.s; e = right.po
 // =============================
 
 /// Wrapper type for every AST node.
-/// This enables adding extensible metadata to the AST for tracking things such as source code position.
+/// This enables adding extensible metadata to the AST for tracking things such as source 
+/// code position.  pos tracks position history of this node (when functions expand) with highest level call first
+/// with 
 [<CustomEquality>][<NoComparison>]
-type Node<'T when 'T: equality> = {x: 'T; pos: SourcePosition option}
+type Node<'T when 'T: equality> = {x: 'T; positions: SourcePosition list}
     with
+    member x.Pos = 

```

Comment:
This property might be unused after refactoring the position addition code to use the full list of positions?  Not sure though.

Reply:
Again good observation backed up by experiemental evidence.  Deleted.